lib/IRGen/IRGenSIL.cpp

```diff
@@ -2314,14 +2314,13 @@ void IRGenSILFunction::visitPartialApplyInst(swift::PartialApplyInst *i) {
   
   Explosion llArgs;
 
-  {
-    // Lower the parameters in the callee's generic context.
-    GenericContextScope scope(IGM, i->getOrigCalleeType()->getGenericSignature());
-    for (auto index : indices(args)) {
-      assert(args[index]->getType() == IGM.silConv.getSILType(params[index]));
-      emitApplyArgument(*this, args[index],
-                        IGM.silConv.getSILType(params[index]), llArgs);
-    }
+  // Lower the parameters in the callee's generic context.
```

Comment:
Why is the scope removed here?

Reply:
I need the generic environment down the call stack: in order to map the CanType in context / create the mangled name

Reply:
You should in general pass it down and map out of context yourself instead of relying on GenericContextScope. Like I said I'd like to remove it soon.

Reply:
I actually had a version of the code that did that: felt it looked ugly / not an elegant solution: touching the code in multiple places to pass the information down all the way